gym_electric_motor/callbacks.py

```diff
@@ -46,32 +49,30 @@ def __init__(self, initial_limit_margin=(-0.1,0.1), maximum_limit_margin=(-1,1),
     def set_env(self, env):
         #see docstring of superclass
         #assertions added to check for the right reference generator 
-        if env._reference_generator.__class__ == SwitchedReferenceGenerator:
+        if isinstance(env._reference_generator, SwitchedReferenceGenerator):
             for sub_generator in env._reference_generator._sub_generators:
-                assert issubclass(sub_generator.__class__, SubepisodedReferenceGenerator), ("The AdaptiveLimitMargin does only support the SubepisodedReferenceGenerator as reference generator or "
-                                                                                            "SwitchedReferenceGenerator with SubepisodedReferenceGenerator as all sub reference generators")
+                assert issubclass(type(sub_generator), SubepisodedReferenceGenerator), self.__CLASS_ERROR__
         else:
-            assert issubclass(env._reference_generator.__class__, SubepisodedReferenceGenerator), ("The AdaptiveLimitMargin does only support the SubepisodedReferenceGenerator as reference generator or" 
-                                                                                                 "SwitchedReferenceGenerator with SubepisodedReferenceGenerator as all sub reference generators")
+            assert issubclass(type(env._reference_generator), SubepisodedReferenceGenerator), self.__CLASS_ERROR__
         self._env = env
         #initial image margin added to the reference generator
-        if env._reference_generator.__class__ == SwitchedReferenceGenerator:
+        if isinstance(env._reference_generator, SwitchedReferenceGenerator):
             for sub_generator in self._env._reference_generator._sub_generators:
                 sub_generator._limit_margin = self._limit_margin
         else:
             self._env._reference_generator._limit_margin = self._limit_margin
         
     def on_step_end(self):
         #see docstring of superclass
-        if self._update_time == 0:
+        if self._update_time == 'step':
             self._step += 1
             if self._step % self._update_freq == 0:
                 self._step = 0
                 self._update_limit_margin()     
                 
     def on_reset_end(self):
         #see docstring of superclass
-        if self._update_time == 1:
+        if self._update_time == 'episode':
```

Comment:
Comparing to strings is like one microsecond slower:

~$ python -m timeit "s = 'step'" "if (s == 'step'):" "  pass"
100000000 loops, best of 3: 0.0186 usec per loop
~$ python -m timeit "s = 3" "if (s == 3):" "  pass"
100000000 loops, best of 3: 0.0174 usec per loop